tools/sysctlfix/main.go

```diff
@@ -139,3 +108,38 @@ func main() {
 
 	fmt.Printf("systemd unit '%s' restarted\n", *sysctlUnit)
 }
+
+func writeConfig(overwritesPath string) error {
+	f, err := os.OpenFile(overwritesPath, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		return fmt.Errorf("unable to create cilium sysctl overwrites config: %w\n", err)
+	}
+	defer f.Close()
```

Comment:
Oof, this is subtle. Took me a long time to understand what the difference even was 😅 

Given the timing of `f.Close` is crucial, can we comment on it, or even better, call `f.Sync()` explicitly somewhere?

Reply:
That can't hurt. :+1: